dwh/indicator_lib.py

```diff
@@ -160,10 +94,45 @@ def calc_TX_PVLS(patient_agg_obs: DataFrame, VL_code: str, failure_threshold: in
   num_patients = VL_df.count()
   VL_agg_P = VL_df.select(
       VL_df.sup_VL,
-      F.explode(VL_df.agg_buckets).alias('agg_bucket')).groupBy(
+      F.explode(VL_df._agg_buckets).alias('agg_bucket')).groupBy(
       'sup_VL', 'agg_bucket').agg(
       F.count('*').alias('count')).toPandas().sort_values(
       ['agg_bucket', 'sup_VL'])
   VL_agg_P['ratio'] = VL_agg_P['count']/num_patients
   return VL_agg_P
 
+
+# TODO remove this function once we verify that that pandas.DataFrame based
```

Comment:
Would be nice to aslo consider [spark's koala](https://koalas.readthedocs.io/en/latest/) : That way we get the best of both worlds - you can write indicator logic using panda's API  as well as spark's SQL, etc., while still leveraging multi-node and in-memory processing capabilities of spark

Reply:
Makes sense, thanks for pointing that out